lib/services/settings/mobile_settings_service.dart

```diff
@@ -58,16 +60,24 @@ class MobileSettingsService extends SettingsService {
   }
 
   @override
-  bool get themeDarkMode {
-    var theme = _sharedPreferences.getString('theme') ?? 'dark';
-
-    return theme == 'dark';
+  String get themeMode {
+    var value = _sharedPreferences.getString('theme') ?? ThemeMode.system.name;
```

Comment:
You fetch the stored preference and default to system if not set, which is good.